### PR TITLE
Issue 6189: Added dmSpinlock to dmSDK

### LINKED ISSUE #6189: Add spinlock to dmSDK
It's useful as a more light weight lock.

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/dlib/src/dlib/message.cpp [hunk 1]
```diff
@@ -123,7 +123,7 @@ namespace dmMessage
     struct MessageContext
     {
         dmHashTable64<MessageSocket> m_Sockets;
-        dmSpinlock::lock_t m_Spinlock;
```
Comment (JCash) [inline engine/dlib/src/dlib/message.cpp:126]: We don't use that naming pattern for our types.

**Review by AGulev — DISMISSED** [review]
👍

File: engine/dlib/src/dmsdk/dlib/spinlock.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+     * @name dmSpinlock::Spinlock
+     *
+     */
+
+    /*# initalize spinlock.
+     *
+     * Initialize a Spinlock
+     *
+     * @name dmSpinlock::Init
+     * @param spinlock [type:dmSpinlock::Spinlock*] spinlock to unlock.
```
Comment (britzl) [inline engine/dlib/src/dmsdk/dlib/spinlock.h:46]: unlock -> initialize

**Review by britzl — CHANGES_REQUESTED** [review]
Typo in docs

**Pull request merged**